swarm/src/behaviour.rs

```diff
@@ -299,21 +290,20 @@ pub enum ToSwarm<TOutEvent, TInEvent> {
         event: TInEvent,
     },
 
-    /// Informs the `Swarm` about an address observed by a remote for
-    /// the local node by which the local node is supposedly publicly
-    /// reachable.
+    /// Reports a new candidate for an external address to the [`Swarm`].
     ///
-    /// It is advisable to issue `ReportObservedAddr` actions at a fixed frequency
-    /// per node. This way address information will be more accurate over time
-    /// and individual outliers carry less weight.
-    ReportObservedAddr {
-        /// The observed address of the local node.
-        address: Multiaddr,
-        /// The score to associate with this observation, i.e.
-        /// an indicator for the trusworthiness of this address
-        /// relative to other observed addresses.
-        score: AddressScore,
-    },
+    /// This address could come from a variety of sources:
+    /// - A protocol such as identify obtained it from a remote.
+    /// - The user provided it based on configuration.
+    /// - We made an educated guess based on one of our listen addresses.
+    /// - We established a new relay connection.
+    NewExternalAddrCandidate(Multiaddr),
+
+    /// Indicates to the [`Swarm`] that the provided address is confirmed to be externally reachable.
+    ExternalAddrConfirmed(Multiaddr),
+
+    /// Indicates to the [`Swarm`] that we are no longer externally reachable under the provided address.
+    ExternalAddrExpired(Multiaddr),
```

Comment:
To make sure we are on the same page, one would only emit this event to invalidate previous `ExternalAddrConfirmed`, right? There is no need to invalidate `NewExternalAddrCandidate`. Correct?

Reply:
Yes,

Reply:
```suggestion
    ///
    /// Note that a [`ExternalAddrExpired`] only invalidates a previous [`ExternalAddrConfirmed`].
    /// One does not need to invalidate a [`NewExternalAddrCandidate`].
    ExternalAddrExpired(Multiaddr),
```

Mind adding something along the lines above to avoid confusions in the future?

Reply:
Happy to add something although I'd hope that the naming itself is sufficient. Why would one want to expire a candidate?